src/client/shader.h

```diff
@@ -67,6 +67,8 @@ class IShaderConstantSetter {
 	virtual ~IShaderConstantSetter() = default;
 	virtual void onSetConstants(video::IMaterialRendererServices *services,
 			bool is_highlevel) = 0;
+	virtual void onSetMaterial(const video::SMaterial& material)
+	{ }
```

Comment:
`= 0`?

Reply:
This will force GameGlobalShaderConstantSetter and MainShaderConstantSetter to implement empty onSetMaterial callbacks. If that's ok I'll make the change

Reply:
Nah, that's okay as is.